src/main/java/org/wso2/carbon/module/swiftiso20022/utils/ISOMessageParser.java

```diff
@@ -0,0 +1,122 @@
+        AXIOMXPath xpathExp = new AXIOMXPath(xPath);
+
+        // Setting up XML namespaces for AXIOM SOAP, AppHdr, and Document
+        xpathExp.addNamespace(ConnectorConstants.SOAP_PREFIX, ConnectorConstants.SOAP_NAMESPACE);
+        xpathExp.addNamespace(ConnectorConstants.APPHDR_PREFIX, ConnectorConstants.XML_INPUT_APPHDR_NAMESPACE);
+        xpathExp.addNamespace(ConnectorConstants.DOCUMENT_PREFIX, ConnectorConstants.XML_INPUT_DOCUMENT_NAMESPACE);
+
+        OMNode rootElement = mc.getEnvelope().getBody();
+        return (OMElement) xpathExp.selectSingleNode(rootElement);
+    }
+
+    /**
+     * Retrieve specific list of OMElement from MessageContext according to the XPATH.
+     * @param xPath                 XPath expression to the OMElements
+     * @param mc                    MessageContext which contains the XML 20022 input
+     * @return "List<OMELement>"    List of OMElements
+     * @throws JaxenException
+     */
+    public static List<OMElement> getXMLElementsByPath(String xPath, MessageContext mc) throws JaxenException {
+        AXIOMXPath xpathExp = new AXIOMXPath(xPath);
+
+        // Setting up XML namespaces for AXIOM SOAP, AppHdr, and Document
+        xpathExp.addNamespace(ConnectorConstants.SOAP_PREFIX, ConnectorConstants.SOAP_NAMESPACE);
+        xpathExp.addNamespace(ConnectorConstants.APPHDR_PREFIX, ConnectorConstants.XML_INPUT_APPHDR_NAMESPACE);
+        xpathExp.addNamespace(ConnectorConstants.DOCUMENT_PREFIX, ConnectorConstants.XML_INPUT_DOCUMENT_NAMESPACE);
+
+        OMNode rootElement = mc.getEnvelope().getBody();
+        return xpathExp.selectNodes(rootElement);
+    }
+
+    /**
+     * Read the XML input from MessageContext and return the content of child elements of the parent tag as a String.
+     * @param mc         MessageContext which contains the XML 20022 input
+     * @param xPath      Xpath for the XML contents which needs to be extracted
+     * @return           Child contents as a String element
+     */
+    public static String extractISOMessage(MessageContext mc, String xPath) throws ConnectException,
+            OMException, JaxenException  {
+        OMElement element = getXMLElementByXPath(xPath, mc);
+
+        if (element == null) {
+            // No element to be extracted
+            String errMsg = String.format("Error: %s element not present in the XML", xPath);
+
+            throw new ConnectException(errMsg);
+        }
+
+        return element.toString();
+    }
+
+    /**
+     * Read the ISO message from MessageContext and return the Root element tag name.
+     * @param mc         MessageContext which contains the XML 20022 input
+     * @return           Root XML tag name
+     */
+    public static String getRootXMLElement(MessageContext mc) {
+        try {
+            OMElement rootElement = mc.getEnvelope().getBody().getFirstElement();
+            return rootElement.getLocalName();
+        } catch (NullPointerException e) {
+            return null;
```

Comment:
Usually we don't catch runtime exceptions like NPE and checks !=null in the logic

Reply:
Fixed using Optional in 5f315274d9d171aa9caf2101d47fc8ec05a8e7cb